crates/blockchain/blockchain.rs

```diff
@@ -48,7 +48,11 @@ impl Blockchain {
         }
     }
 
-    pub fn add_block(&self, block: &Block) -> Result<(), ChainError> {
+    pub fn add_block(
```

Comment:
I was thinking something in the lines of adding `add_blocks`, which only commits the trie once.

Then you can trivially add 
`fn add_block(block) { add_blocks(vec![block])}`

Thoughts?